src/Sylius/Bundle/CoreBundle/Tests/Form/Type/CatalogPromotion/CatalogPromotionActionTypeTest.php

```diff
@@ -0,0 +1,246 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\CoreBundle\Tests\Form\Type\CatalogPromotion;
+
+use Prophecy\Argument;
+use Prophecy\Prophecy\ObjectProphecy;
+use Prophecy\Prophecy\ProphecyInterface;
+use Sylius\Bundle\CoreBundle\Form\Type\CatalogPromotionAction\ChannelBasedFixedDiscountActionConfigurationType;
+use Sylius\Bundle\CoreBundle\Form\Type\ChannelCollectionType;
+use Sylius\Bundle\PromotionBundle\Form\Type\CatalogPromotionAction\PercentageDiscountActionConfigurationType;
+use Sylius\Bundle\PromotionBundle\Form\Type\CatalogPromotionActionType;
+use Sylius\Component\Addressing\Model\CountryInterface;
+use Sylius\Component\Channel\Repository\ChannelRepositoryInterface;
+use Sylius\Component\Core\Model\ChannelInterface;
+use Sylius\Component\Currency\Model\CurrencyInterface;
+use Sylius\Component\Promotion\Model\CatalogPromotionAction;
+use Sylius\Component\Promotion\Model\CatalogPromotionActionInterface;
+use Symfony\Component\Form\Extension\Validator\ValidatorExtension;
+use Symfony\Component\Form\PreloadedExtension;
+use Symfony\Component\Form\Test\TypeTestCase;
+use Symfony\Component\Validator\Validation;
+
+final class CatalogPromotionActionTypeTest extends TypeTestCase
+{
+    private CatalogPromotionActionInterface $fixedDiscount;
+
+    private CatalogPromotionActionInterface $percentageDiscount;
+
+    private ProphecyInterface|CountryInterface $channel;
+
+    private ObjectProphecy $channelRepository;
+
+    protected function setUp(): void
+    {
+        $this->channelRepository = $this->prophesize(ChannelRepositoryInterface::class);
+
+        $currency = $this->prophesize(CurrencyInterface::class);
+        $currency->getCode()->willReturn('$');
+
+        $channel = $this->prophesize(ChannelInterface::class);
+        $channel->getCode()->willReturn('WEB_US');
+        $channel->getName()->willReturn('United States');
+        $channel->getBaseCurrency()->willReturn($currency->reveal());
+        $this->channel = $channel;
+
+        $percentageDiscount = new CatalogPromotionAction();
+        $percentageDiscount->setType('percentage_discount');
+        $percentageDiscount->setCatalogPromotion(null);
+        $percentageDiscount->setConfiguration(['amount' => 0.1]);
+        $this->percentageDiscount = $percentageDiscount;
+
+        $fixedDiscount = new CatalogPromotionAction();
+        $fixedDiscount->setType('percentage_discount');
+        $fixedDiscount->setCatalogPromotion(null);
+        $fixedDiscount->setConfiguration(['WEB_US' => ['amount' => 10]]);
+        $this->fixedDiscount = $fixedDiscount;
```

Comment:
It would be better to have two custom setup methods, that we can call at the beginning of test - there is not point in creation both of these actions at the same time